src/handles/handleDiagnostic.ts

```diff
@@ -83,18 +83,25 @@ async function handleLinter (
     return diagnostics
   }
   try {
+    const fpath = VscUri.parse(textDocument.uri).path;
     const workDir = await findWorkDirectory(
-      VscUri.parse(textDocument.uri).path,
+      fpath,
       rootPatterns
     )
     const cmd = await findCommand(command, workDir)
+    const tmp = (args as any[] || []).map(arg => {
+      if (/%file/.test(arg)) {
+        return arg.replace(/%file/g, fpath.toString())
+      }
+      return arg
+    })
```

Comment:
change this code to https://github.com/iamcco/diagnostic-languageserver/blob/master/src/common/util.ts#L23 seems make sense

Reply:
Humm, then we have to modify the executeFile function. Since there is no way to access the textDocument.uri from that function right now.

Reply:
It's OK to change to function to accept `Object` params.